python/taichi/lang/matrix.py

```diff
@@ -1136,43 +1138,68 @@ def field(cls,
         impl.get_runtime().matrix_fields.append(entries)
 
         if shape is None:
-            assert offset is None, "shape cannot be None when offset is being set"
-
-        if shape is not None:
+            if offset is not None:
+                raise TaichiSyntaxError(
+                    'Shape cannot be None when offset is set')
```

Comment:
Super nit: `shape`, `offset` are param names, so let's keep it all lowercases.

Reply:
Fixed